tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -248,10 +256,16 @@ struct UnsortedSegmentSumFunctor<CPUDevice, T, Index> {
 
 // Similar to SegmentReductionOp but can handle unsorted segment definitions and
 // specifying size of output.
-template <typename Device, class T, class Index>
-class UnsortedSegmentSumOp : public OpKernel {
+template <typename Device, class T, class Index,
+          void (*unsortedSegmentInitializationOp)(
```

Comment:
I dont like passing these things as template arguments.  Can we make these things functors, and pass the proper functor in to the base classes' Compute method?

class UnsortedSegmentSumOp : public UnsortedSegmentBase

  void Compute(...) {
      // Pass in UnsortedSegmentSum functor here.
      UnsortedSegmentBase::Compute(functor);
   }